src/atomic/SDL_spinlock.c

```diff
@@ -160,8 +142,22 @@ SDL_bool SDL_AtomicTryLock(SDL_SpinLock *lock)
     }
     return res;
 #else
-#error Please implement for your platform.
-    return SDL_FALSE;
+    /* Terrible terrible damage */
+    static SDL_Mutex *_spinlock_mutex;
+
+    if (!_spinlock_mutex) {
+        /* Race condition on first lock... */
+        _spinlock_mutex = SDL_CreateMutex();
+    }
```

Comment:
On architectures with no atomic operations available, the way libdbus handles this is to assume that all platforms are either Windows or pthread, and Windows always has atomics, therefore if we get here we can use pthread; and pthread lets you initialize an unlocked mutex statically:

```
static pthread_mutex_t _spinlock_mutex = PTHREAD_MUTEX_INITIALIZER;

if (pthread_mutex_lock (&_spinlock_mutex) != 0)
  /* error, look at errno for details */

... do stuff under the lock ...

if (pthread_mutex_unlock (&_spinlock_mutex) != 0)
  /* error, look at errno for details */
```

I realise SDL is portable to more platforms than libdbus, and some of them are awful; but would it be reasonable to assume/require that each of them has either atomic operations, or a way to declare a pre-initialized mutex statically like that? That would eliminate this race condition.

Reply:
To test this sort of thing, a convenient example of a platform with no built-in atomic operations is armv5, as used in Debian's `armel` architecture.

Reply:
That's a reasonable solution, we could have three implementations there, one for Windows, one for pthreads, and the existing code.